src/Bicep.RegistryModuleTool/Proxies/ProcessProxy.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+
+namespace Bicep.RegistryModuleTool.Proxies
+{
+    internal sealed class ProcessProxy : IProcessProxy
+    {
+        public (int exitCode, string standardOutput, string standardError) Start(string executablePath, string arguments)
+        {
+            var process = new Process();
+
+            process.StartInfo.FileName = executablePath;
+            process.StartInfo.Arguments = arguments;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.UseShellExecute = false;
+
+            process.Start();
+
+            // TODO: fix deadlock.
```

Comment:
>deadlock [](http://example.com/codeflow?start=25&length=8)

The easiest way is to enable the events that notify you about error and output lines and consume from them. That prevents the StdOut/StdErr buffer(s) from filling up and deadlocking the app.

Reply:
I might have an example of this somewhere in the repo.

Reply:
I didn't find the example in the repo, but I ended up solving it by reading the two buffers in different threads, which is mentioned in the API docs.

Reply:
Oh interesting. Didn't know this was the other option. This works too.

Mine was this btw: [https://cs.github.com/Azure/azure-sdk-for-net/blob/8b55e111d195935f6a7db695b5197750cfb03856/sdk/identity/Azure.Identity/src/ProcessRunner.cs?q=OutputDataReceived#L80](https://cs.github.com/Azure/azure-sdk-for-net/blob/8b55e111d195935f6a7db695b5197750cfb03856/sdk/identity/Azure.Identity/src/ProcessRunner.cs?q=OutputDataReceived#L80)